Source/player.cpp

```diff
@@ -622,13 +614,12 @@ void DropHalfPlayersGold(int pnum)
 	}
 	auto &player = Players[pnum];
 
-	int hGold = player._pGold / 2;
-
-	hGold = DropGold(pnum, hGold, true);
-	if (hGold > 0)
-		DropGold(pnum, hGold, false);
+	int remainingGold = DropGold(player, player._pGold / 2, true);
+	if (remainingGold > 0) {
+		DropGold(player, remainingGold, false);
+	}
 
-	player._pGold -= hGold;
+	player._pGold /= 2;

```

Comment:
Looks like you fixed an issue with how remaining gold was calculated here :)

Reply:
yeah, the sum of gold in the inventory value was getting set to a value that didn't actually represent what the player was holding as hGold was being reused to determine how much additional gold to drop from the second pass. I don't think it mattered much in practice as the value is only used in shops and players can only die in the dungeon (normally :D).